### ISSUE TITLE
pipeline compatibility issue

### ISSUE THREAD

**Issue by coreyfarrell** [mafintosh/streamx#38]
In certain situations `pipeline` does not execute the `done` callback.  In these cases switching the promise interface never completes.

### Setup
```
npm i readable-stream@3.6.0 concat-stream@2.0.0 streamx@2.10.0
```

### Test case
```js
const {Readable, pipeline} = require(process.argv[2] || 'stream');
const concat = require('concat-stream');

pipeline(
	Readable.from('text'),
	concat(result => {
		console.log('concat result:', result);
	}),
	() => {
		console.log('pipeline finished');
	}
)
```

Running this script with no argument prints:
```
concat result: text
pipeline finished
```

The same success case is produced if you run the program with `stream` or `readable-stream` as the argument.

Running this script with `streamx` as an argument only prints `concat result: text` then the event loop ends.

**Comment by phated** [comment]
@coreyfarrell this is blocking gulp upgrades, right? cc @mafintosh

**Comment by coreyfarrell** [comment]
@phated I think it would block any releases.  I would like to ensure gulp has adequate testing with different mixtures of streamx combined with node.js native streams.  I've got tests working in `to-through` using `streamx`, just without using `streamx.pipeline` in the tests.

**Comment by phated** [comment]
> just without using streamx.pipeline in the tests.

Hmmm, `streamx.pipeline` will be the recommended pattern for end users to combine streams. A lot of people have already migrated to `stream.pipeline` or `pump`

**Comment by mafintosh** [comment]
Thanks! Will fix

**Comment by mafintosh** [comment]
fixed in 2.13.3 (compat with very old streams)
